libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -357,6 +359,7 @@ public void startNavigation(@NonNull DirectionsRoute directionsRoute) {
         .previousRouteDistancesCompleted(0)
         .startTimestamp(new Date())
         .rerouteCount(0)
+        .locationEngineName(locationEngine.getClass().getSimpleName())
```

Comment:
Be careful because this is executed before `setLocationEngine` so it always has the original value (`LostLocationEngine`).

Reply:
Maybe adding it to the `SessionState` is not the best option.